steel/docs/steel-commitments.md

```diff
@@ -0,0 +1,138 @@
+require(Steel.validateCommitment(journal.commitment), "Invalid commitment");
+```
+
+The [Steel library](https://github.com/risc0/risc0-ethereum/blob/main/contracts/src/steel/Steel.sol) contains the function `validateCommitment` :
+
+```javascript
+function validateCommitment(Commitment memory commitment) internal view returns (bool) {
+	(uint240 blockID, uint16 version) = Encoding.decodeVersionedID(commitment.blockID);
+	if (version == 0) {
+		return validateBlockCommitment(blockID, commitment.blockDigest);
+	} else if (version == 1) {
+		return validateBeaconCommitment(blockID, commitment.blockDigest);
+	} else {
+		revert InvalidCommitmentVersion();
+	}
+}
+```
+
+## Validation of Steel Commitments
+
+Steel supports two methods of commitment validation (see `validateCommitment` in [Steel.sol]); This validation on-chain is essential to ensure that the proof accurately reflects the correct blockchain state.
+
+1. Block hash commitment
+
+```solidity
+/// @notice Validates if the provided block commitment matches the block hash of the given block number.
+/// @param blockNumber The block number to compare against.
+/// @param blockHash The block hash to validate.
+/// @return True if the block's block hash matches the block hash, false otherwise.
+function validateBlockCommitment(uint256 blockNumber, bytes32 blockHash) internal view returns (bool) {
+	if (block.number - blockNumber > 256) {
+		revert CommitmentTooOld();
+	}
+	return blockHash == blockhash(blockNumber);
+}
+```
+
+This method uses the `blockhash` opcode to commit to a block hash that is no more than 256 blocks old. With Ethereum's 12-second block time, this provides a window of about 50 minutes to generate the proof and ensure that the validating transaction is contained in a block. This approach is ideal for most scenarios, including complex computations, as it typically provides sufficient time to generate the proof.
+
+2. Beacon Block Root Commitment
+
+```solidity
+/// @notice Validates if the provided beacon commitment matches the block root of the given timestamp.
+/// @param blockTimestamp The timestamp to compare against.
+/// @param blockRoot The block root to validate.
+/// @return True if the block's block root matches the block root, false otherwise.
+function validateBeaconCommitment(uint256 blockTimestamp, bytes32 blockRoot) internal view returns (bool) {
+	if (block.timestamp - blockTimestamp > 12 * 8191) {
+		revert CommitmentTooOld();
+	}
+	return blockRoot == Beacon.blockRoot(blockTimestamp);
+}
+```
+
+The second method allows validation using the [EIP-4788] beacon roots contract. This technique extends the time window in which the proof can be validated on-chain to just over a day, making it suitable for scenarios requiring more extensive computation. It requires access to a beacon API endpoint and can be enabled by calling `EvmEnv::into_beacon_input`. However, this approach is specific to Ethereum Steel proofs and depends on the implementation of EIP-4788.
```

Comment:
It's not so much about the amount of computation, since with Bonsai and similar we can get a lot of computation proven within the 50 minute window (we've demonstrated proving ~50B cycles in ~30 minutes). But there are other reasons, and folks who need a longer window will know it.
```suggestion
The second method allows validation using the [EIP-4788] beacon roots contract. This technique extends the time window in which the proof can be validated on-chain to just over a day. It requires access to a beacon API endpoint and can be enabled by calling `EvmEnv::into_beacon_input`. However, this approach is specific to Ethereum (L1) Steel proofs and depends on the implementation of EIP-4788.
```